### ISSUE TITLE
LayerNormalization wrong "axis" parameter coded

### ISSUE THREAD

**Issue by CarlosNacher** [keras-team/keras#18231]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): YES
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): 
- TensorFlow installed from (source or binary): I don't know. In Colab
- TensorFlow version (use command below): v2.12.0-rc1-12-g0db597d0d75 2.12.0
- Python version: 3.10.12
- GPU model and memory: None

**Describe the problem**.

Layer Normalization with axis = -1 normalises calculating the mean of each pixel across the channels, instead of the mean of all pixels in a example across the channels (HxW). If you set axis=-1, then there are as many parameters as channels. But if you test with a generated array, you see that it does not calculate n_channels averages, but actually calculates an average for each pixel (HxW).

**Describe the current behavior**.

Above

**Describe the expected behavior**.

I don't find the sense of having n_channels parameters if when you test LN, you see that there are HxW means calculated. I've put all this in a notebook. See below (is more explained).

**Standalone code to reproduce the issue**.

In this Notebook I do some tests: https://colab.research.google.com/drive/1EMXLaUkCLO-J_svFOJ_fMkMCoUFQ4Ewu?usp=sharing

**Comment by CarlosNacher** [comment]
> @CarlosNacher, As per the docs, Relation to Layer Normalization: If the number of groups is set to 1, then this operation becomes **nearly identical** to Layer Normalization. https://www.tensorflow.org/api_docs/python/tf/keras/layers/GroupNormalization https://www.tensorflow.org/api_docs/python/tf/keras/layers/Normalization
> 
> Thank you!


Hello, thank you very much for your response!
Yes, but what I would really like to highlight is the fact that by choosing axis = -1, depending on the layer (`trainable_variables` there are as many parameters as there are channels, but depending on the effect (apply the layer to an image, tested on the notebook) the image is normalized per pixel so there are actually more averages than the number of channels. I may be missing something, could you clarify this please? thank you so much!

**Comment by Avelina9X** [comment]
@CarlosNacher

Layer Norm is working correctly. Layer Norm with `axis=-1` normalizes per pixel for 4D tensors and per timestep for 3D tensors. If you are looking for normalization over dimensions 1 and 2 for 4D tensors (i.e. across width and height per channel) then you should look into instance normalization which you could easily implement yourself, or use the layer provided in TFA.
